src/ui/win32/DialogBase.cpp

```diff
@@ -19,6 +19,10 @@ DialogBase::~DialogBase() noexcept
 {
     if (m_hWnd)
     {
+        // remove the pointer to the instance from the HWND so the StaticDialogProc doesn't try to redirect
+        // the WM_DESTROY message through the instance we're trying to destroy.
+        ::SetWindowLongPtr(m_hWnd, DWLP_USER, 0L);
```

Comment:
In XP, the destructor was being called before the window was destroyed, which caused line 26 to destroy the window. The window's OnDestroy handler called back into the Presenter to destroy the singleton instance, which was already being destroyed by the destructor, causing the crash. This prevents the message from being forwarded to the dialog that's already being destroyed.